### ISSUE TITLE
Async Support | Search Tools

### ISSUE THREAD

**Issue by jordanparker6** [langchain-ai/langchain#5011]
### Feature request

None of the search tools have async support.

### Motivation

Async calls are being used langchain setup with FastAPI. Not having async support for these tools blocks there use.

### Your contribution

Happy to help out where needed.

**Comment by somethingwentwell** [comment]
I faced the same problem and created a custom tool to handle it. here's the code for your reference.

```
import os
from langchain.agents import Tool
from dotenv import load_dotenv
import aiohttp

async def search_bing(query):
    subscription_key = str(os.getenv("BING_SUBSCRIPTION_KEY"))
    endpoint = str(os.getenv("BING_SEARCH_URL"))
    headers = {"Ocp-Apim-Subscription-Key": subscription_key}
    params = {"q": query, "count": 3}
    async with aiohttp.ClientSession() as session:
        async with session.get(endpoint, headers=headers, params=params) as response:
            response.raise_for_status()
            return await response.json()

a_bing_tool = Tool(
        name = "search",
        func=search_bing,
        description="useful for when you need to answer questions about current events",
        coroutine=search_bing
    )
```

**Comment by oneryalcin** [comment]
I think a more general purpose solution available to make any tool async (not fully native async but does the purpose) Use asyncer library to convert a sync function to async. For example DuckDuckGo search tool doesn't have a `_arun` function

but it is pretty easy to implement one

```python
from asyncer import asyncify

class DuckDuckGoSearchRun(BaseTool):
    """Tool that queries the DuckDuckGo search API."""

    name = "duckduckgo_search"
    description = ("...")
    )
    
    ...

    def _run(...)

    def _arun(self, *args: Any, **kwargs: Any) -> Any:
        """Async version of the tool"""
        return asyncify(self._run, cancellable=True)(*args, **kwargs)
```

**Comment by oneryalcin** [comment]
Please disregard as this has been addressed in https://github.com/langchain-ai/langchain/pull/10058 recently. No need anymore

**Comment by sowsan** [comment]
> Please disregard as this has been addressed in #10058 recently. No need anymore

Is there an example of how to use this async tool? I have used the arun and run but still it is calling run though I run this tool with the agent using await

**Comment by weijie7** [comment]
I have the same problem. My agent chain needs to call the function several times for different parameters, and it doesn't seem to perform that asynchronously. I'm using llm_agent.ainvoke function, and defined my tool function to have _run and _arun properly. Doesn't seem to work. It is still calling tool sequencially
